### PR TITLE
DBPW - Revert AutoMTLS

### PR DESCRIPTION
**Pull request by pcman312**
# Description
AutoMTLS somehow screwed up the tests around using v5. The tests were passing previously because of partial vendoring in https://github.com/hashicorp/vault/pull/10008 and broke when the remaining vendoring came in https://github.com/hashicorp/vault/pull/9794

Since a bunch of things have merged since #10008 and because we don't want to get into a strange state where vendoring is janky we're disabling AutoMTLS which appears to fix the tests.

### CODE REVIEW

**Review by calvn — APPROVED** [review]
Ran `TestPlugin_lifecycle` locally and it passed. Might want to wait for CI to finish to see if it catches any regressions though.

**Comment by pcman312** [comment]
Agreed - I'm waiting for CI to be green (or at least completely unrelated) before merging. I'm also running all tests associated with the database plugin locally (`richgo test -v github.com/hashicorp/vault/sdk/database/... github.com/hashicorp/vault/builtin/logical/database/... github.com/hashicorp/vault/sdk/helper/pluginutil -failfast`)

**Review by tvoran — APPROVED** [review]

**Pull request merged**
